scripts/convert_flux_control_lora_to_diffusers.py

```diff
@@ -0,0 +1,393 @@
+                    f"time_text_embed.guidance_embedder.linear_2.{diffusers_lora_key}.bias"
+                ] = original_state_dict.pop(f"guidance_in.out_layer.{lora_key}.bias")
+
+        # context_embedder
+        converted_state_dict[f"context_embedder.{diffusers_lora_key}.weight"] = original_state_dict.pop(
+            f"txt_in.{lora_key}.weight"
+        )
+        if f"txt_in.{lora_key}.bias" in original_state_dict.keys():
+            converted_state_dict[f"context_embedder.{diffusers_lora_key}.bias"] = original_state_dict.pop(
+                f"txt_in.{lora_key}.bias"
+            )
+
+        # x_embedder
+        converted_state_dict[f"x_embedder.{diffusers_lora_key}.weight"] = original_state_dict.pop(
+            f"img_in.{lora_key}.weight"
+        )
+        if f"img_in.{lora_key}.bias" in original_state_dict.keys():
+            converted_state_dict[f"x_embedder.{diffusers_lora_key}.bias"] = original_state_dict.pop(
+                f"img_in.{lora_key}.bias"
+            )
+
+    # double transformer blocks
+    for i in range(num_layers):
+        block_prefix = f"transformer_blocks.{i}."
+
+        for lora_key, diffusers_lora_key in zip(["lora_A", "lora_B"], ["lora_A", "lora_B"]):
+            # norms
+            converted_state_dict[f"{block_prefix}norm1.linear.{diffusers_lora_key}.weight"] = original_state_dict.pop(
+                f"double_blocks.{i}.img_mod.lin.{lora_key}.weight"
+            )
+            if f"double_blocks.{i}.img_mod.lin.{lora_key}.bias" in original_state_dict.keys():
+                converted_state_dict[
+                    f"{block_prefix}norm1.linear.{diffusers_lora_key}.bias"
+                ] = original_state_dict.pop(f"double_blocks.{i}.img_mod.lin.{lora_key}.bias")
+
+            converted_state_dict[
+                f"{block_prefix}norm1_context.linear.{diffusers_lora_key}.weight"
+            ] = original_state_dict.pop(f"double_blocks.{i}.txt_mod.lin.{lora_key}.weight")
+            if f"double_blocks.{i}.txt_mod.lin.{lora_key}.bias" in original_state_dict.keys():
+                converted_state_dict[
+                    f"{block_prefix}norm1_context.linear.{diffusers_lora_key}.bias"
+                ] = original_state_dict.pop(f"double_blocks.{i}.txt_mod.lin.{lora_key}.bias")
+
+            # Q, K, V
+            if lora_key == "lora_A":
+                sample_lora_weight = original_state_dict.pop(f"double_blocks.{i}.img_attn.qkv.{lora_key}.weight")
+                converted_state_dict[f"{block_prefix}attn.to_v.{diffusers_lora_key}.weight"] = torch.cat(
+                    [sample_lora_weight]
+                )
+                converted_state_dict[f"{block_prefix}attn.to_q.{diffusers_lora_key}.weight"] = torch.cat(
+                    [sample_lora_weight]
+                )
+                converted_state_dict[f"{block_prefix}attn.to_k.{diffusers_lora_key}.weight"] = torch.cat(
+                    [sample_lora_weight]
+                )
```

Comment:
Very important to take not of this conversion.

The original repository calculates QKV (3 * hidden_size) + MLP (4 * hidden_size) with a single nn.Linear and later splits them. With LoRA on this nn.Linear, it is not trivial to remap the lora_A and lora_B layers because of the different dimensions. lora_A has shape `rank, hidden_size`, while lora_B has shape `7 * hidden_size, rank`. Splitting lora_B is trivial, but lora_A needs to be the same for all QKV and MLP.

I think it is mathematically equivalent but if someone could give another look, it would be super helpful!

-----

While writing this comment, I just realized that I made a mistake here and did not account for the MLP lora. Will add immediatedly